uaclient/entitlements/base.py

```diff
@@ -225,6 +225,10 @@ def process_contract_deltas(
         delta_entitlement = deltas.get('entitlement', {})
         transition_to_unentitled = bool(delta_entitlement == util.DROPPED_KEY)
         if not transition_to_unentitled:
+            if delta_entitlement:
+                delta_overrides = util.apply_series_overrides(
+                    deltas, copy_dict=False)
```

Comment:
I should have mentioned context on this change. I would have preferred to always copy the dict because I don't to rely on internal side-effects that mutate params passed to a function. The issue is that util.DROPPED_KEY is an object() that we create when processing contracts deltas. If we copy when processing deltas, the copied values which used to be util.DROPPED_KEY are now !=  util.DROPPED_KEY so delta processing logic which tries to match util.DROPPED_KEY doesn't match anymore.  I couldn't think of another approach (other than making util.DROPPED_KEY a known-unique-string) which would allow us to copy the delta dictionary and still match util.DROPPED_KEY. We could just make this function always mutate, as any of the callsites can/should cope with the mutated dict just fine.

Reply:
An alternative would be to subvert deepcopy specifically for our sentinel value:

```py
class Sentinel:
    def __deepcopy__(self, _memo):
        return self
```

Note that the memory addresses are the same:

```py
>>> DROPPED_KEY = Sentinel()
>>> {'a': DROPPED_KEY}                                                                                                                                                                                         
{'a': <__main__.Sentinel at 0x7f823056f2b0>}
>>> copy.deepcopy({'a': DROPPED_KEY})                                                                                                                                                                          
{'a': <__main__.Sentinel at 0x7f823056f2b0>}
>>> copy.deepcopy({'a': DROPPED_KEY}) == {'a': DROPPED_KEY}
True
```

Reply:
As you said in other comments, all of our callsites are using this object for a fairly short period of time, so mutating it won't be likely to cause us significant issues becase we have a lack of complexity for referencing this object. This is a good alternative suggestion if we were persisting and referencing the object across many callsites in the future. Let's avoid the added work of performing the copy for this. Thanks for the thoughts here!